Ensure consistency in the entire spec (#39)
ETC-A/etca-spec: README.md

# ETCa
Extensible Turing Complete Architecture

This repo is for designing an ISA that has a simple base and multiple extensions that can be added ontop. It's primary goal is to be easy to build in the game
[Turing Complete](https://turingcomplete.game) after the second architecture is completed. Several other goals include being educational and being practical for a real CPU.

An extensible assembler built for this architecture is [here](https://github.com/ETC-A/etca-asm)

## Terminology

_Must_: Specifies that a specific behavior is required by the specification.

_Should_: Specifies that a specific behavior is recommended but not required.

_May_: Specifies that a specific behavior is allowed and not encouraged or discouraged.

_Illegal_: Specifies that a specific behavior or instruction is **NOT** allowed in a compliant implementation and must be handled as specified by the ISA.

_Reserved_: Specifies that the behavior or instruction will be specified in a future extension or feature and should be treated as _illegal_ until it is both specified and implemented.

> _Unspecified_: Specifies that the behavior or instruction is intentionally unconstrained and may be freely handled any way an implementation chooses.

Reviewer: Maybe it should be mentioned here that features could potentially specified previously unspecified behavior?

Author: Good catch. Fixed.

# ETCa
Extensible Turing Complete Architecture

> This repo is for designing an ISA that has a simple base and multiple extensions that can be added ontop. It's primary goal is to be easy to build in the game

Reviewer 2: Are we going for grammar fixes here?
```suggestion
This repo is for designing an ISA that has a simple base and multiple extensions that can be added on top. Its primary goal is to be easy to build in the game
```
"It's" means "it is." Confusing, but "its" is possessive and "it's" is not.

[Turing Complete](https://turingcomplete.game) after the second architecture is completed. Several other goals include being educational and being practical for a real CPU.

> An extensible assembler built for this architecture is [here](https://github.com/ETC-A/etca-asm)

Reviewer 2: ```suggestion
An extensible assembler built for this ISA is [the etca-asm repository](https://github.com/ETC-A/etca-asm) in this organization.
```

## Terminology

_Must_: Specifies that a specific behavior is required by the specification.

_Should_: Specifies that a specific behavior is recommended but not required.

> _May_: Specifies that a specific behavior is allowed and not encouraged or discouraged.

Reviewer 2: These are not quite right: [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt)

Reviewer: These sound like simplifications, but not like they are fundamentally wrong (except maybe `May`). Maybe just saying these are summaries and linking the RFC is a good idea.

Author: I'll be honest, I didn't know where the standard for this was defined. I just went off memory from what I've seen. I agree that `May` should be fixed and the RFC should be linked.

_Illegal_: Specifies that a specific behavior or instruction is **NOT** allowed in a compliant implementation and must be handled as specified by the ISA.

> _Reserved_: Specifies that the behavior or instruction will be specified in a future extension or feature and should be treated as _illegal_ until it is both specified and implemented.

Reviewer 2: Didn't we decide on something more complex here? I was under the impression that it was UB, unless the interrupts extension was available, in which case it was required to cause a `#UD` exception.

Author: In the base ISA, both illegal and reserved are undefined behavior. The interrupt extension mandates that illegal (and by extension reserved) causes an interrupt.
